app/workers/shift_survey_worker.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class ShiftSurveyWorker
+  include Sidekiq::Worker
+  include Rails.application.routes.url_helpers
+
+  def perform 
+    send_shift_surveys
+  end
+
+  private
+ 
+  def send_shift_surveys
+    # Possible Refactor: Get final shift for need per user, send survey for that shift if completed
+      # need.shifts.order(:start_at).group_by(&:user_id)
+      
+    # Shifts that do not have surveys sent in the past day
+    shifts = Shift.where.not('exists (?)', ShiftSurvey.where('shift_surveys.shift_id = shifts.id'))
+    .where('start_at > ?', 1.day.ago)
+    shifts.each do |shift|
+      # Get all shifts for user for the same need
+      user_need_shifts = shift.user.shifts.where(need_id: shift.need_id)
+      remaining_user_need_shifts = user_need_shifts.where('start_at > ?', shift.start_at).where('need_id = ?', shift.need_id)
```

Comment:
I do not believe we need to scope by `need_id` here because this collection is already scoped by `need_id` on the line above.

```suggestion
      remaining_user_need_shifts = user_need_shifts.where('start_at > ?', shift.start_at)
```